sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlockBlobAsyncClient.java

```diff
@@ -392,21 +395,29 @@ private String getBlockID() {
      *         lease on the blob.
      *
      * @return
-     *      A reactive response containing the list of blocks.
+     *      A reactive response containing the list of blocks. This response can be consumed either by items listed or
+     *      by pages of the response. See {@link PagedFlux} for more information.
      */
-    public Flux<BlockItem> listBlocks(BlockListType listType,
+    public PagedFlux<BlockItem> listBlocks(BlockListType listType,
                                       LeaseAccessConditions leaseAccessConditions) {
-        return postProcessResponse(this.azureBlobStorage.blockBlobs().getBlockListWithRestResponseAsync(
-            null, null, listType, snapshot, null, null, null,
-            leaseAccessConditions, Context.NONE))
-            .map(ResponseBase::value)
-            .flatMapMany(bl -> {
-                Flux<BlockItem> committed = Flux.fromIterable(bl.committedBlocks())
-                    .map(block -> new BlockItem(block, true));
-                Flux<BlockItem> uncommitted = Flux.fromIterable(bl.uncommittedBlocks())
-                    .map(block -> new BlockItem(block, false));
-                return Flux.concat(committed, uncommitted);
-            });
+
+        return new PagedFlux<>(
+            () -> postProcessResponse(this.azureBlobStorage.blockBlobs().getBlockListWithRestResponseAsync(
+                    null, null, listType, snapshot, null, null, null,
+                    leaseAccessConditions, Context.NONE))
+                .map(response -> new PagedResponseBase<>(
+                    response.request(),
+                    response.statusCode(),
+                    response.headers(),
+                    Stream.concat(
+                            response.value().committedBlocks().stream()
+                                .map(committedBlock -> new BlockItem(committedBlock, true)),
+                            response.value().uncommittedBlocks().stream()
+                                .map(uncommittedBlock -> new BlockItem(uncommittedBlock, false)))
+                        .collect(Collectors.toList()),
+                    null /* nextLink */,
+                    response.deserializedHeaders())),
+            (marker) -> null);
```

Comment:
This is a single response Flux, this is a current gap in our Flux and PagedFlux returns as it doesn't fit either cleanly. This will be a discussion for Preview 3

Reply:
If other list APIs have a way to do pagination, then customer would expect this one has pagination as well. It is better to have javadoc to make it clear.